internal/pkg/aws/sessions/errors.go

```diff
@@ -24,3 +24,37 @@ func (e *errMissingRegion) RecommendActions() string { // implements new actionR
 - Alternatively, you can run %s to set the environment variable.
 More information: https://aws.github.io/copilot-cli/docs/credentials/`, color.HighlightCode("export AWS_REGION=<application region>"))
 }
+
+type errCredProviderTimeout struct {
+	profile string
+}
+
+// Implements error interface.
+func (e *errCredProviderTimeout) Error() string {
+	return "retrieving credentials timeout"
+}
+
+// RecommendActions returns recommended actions to be taken after the error.
+// Implements main.actionRecommender interface.
+func (e *errCredProviderTimeout) RecommendActions() string {
+	var msg string
+	if e.profile == "default" {
```

Comment:
it's not just the the "default" profile, we want to display this message if for example they use env vars as their credentials but it's misconfigured.

```go
msg := `It looks like your credential settings are misconfigured or missing:
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
- We recommend including your credentials in the shared credentials file.
- Alternatively, you can also set credentials through
	* Environment Variables
	* EC2 Instance Metadata (credentials only)
More information: https://aws.github.io/copilot-cli/docs/credentials/`

if e.profile != "" {
   msg = fmt.Sprintf(`It looks like your profile [%s] is misconfigured or missing:
https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
- We recommend including your credentials in the shared credentials file.
- Alternatively, you can also set credentials through 
	* Environment Variables
	* EC2 Instance Metadata (credentials only)
More information: https://aws.github.io/copilot-cli/docs/credentials/`, e.profile)
}

return msg
```

Reply:
Okay I get it now.